test/functional/apps/discover/_shared_links.js

```diff
@@ -20,6 +20,11 @@ bdd.describe('shared links', function describeIndexTests() {
 
   bdd.before(function () {
     baseUrl = PageObjects.common.getHostPort();
```

Comment:
But I'd still have to use `.getHostPort()` to know if I need the `.getHost()` or not.

Or I could use some other method to get the kibana port first from the config, and then either `.getHostPort()` or `.getHost()`.

So either this;
```
    baseUrl = PageObjects.common.getHostPort();
    if (baseUrl.includes(':80') || baseUrl.includes(':443') {
        baseUrl = PageObjects.common.getHost();
    }
```

or
```
if (config.kibana.port === '443' || config.kibana.port === '80') {
        baseUrl = PageObjects.common.getHost();
} else {
        baseUrl = PageObjects.common.getHostPort();
} 
```